torch/lib/c10d/ProcessGroupMPI.hpp

```diff
@@ -111,10 +120,12 @@ class ProcessGroupMPI : public ProcessGroup {
 
     void abort() override;
 
+   virtual std::vector<at::Tensor> result() override;
+
    protected:
     void populateException();
 
-    at::Tensor tensor_;
+    const std::vector<at::Tensor>* outputTensors_;
```

Comment:
Looks like both `AsyncWork` and `WorkMPI` are user-facing Work implementations. Why they own `outputTensors` in different way?